worker-api/src/main/java/com/hpe/caf/api/worker/NotIntendedTaskMessageForwardingEvaluator.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015-2021 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.api.worker;
+
+import com.hpe.caf.api.Codec;
+import java.util.Map;
+
+/**
+ * A worker should implement this interface if it is capable of evaluating task messages that are not intended for it (because the
+ * worker's input queue is different from the target/to queue in the task message), and deciding whether they are to be discarded,
+ * executed or forwarded.
+ */
+public interface NotIntendedTaskMessageForwardingEvaluator
```

Comment:
I know I'm making basically the same comment here again but I think we have the terminology wrong in this PR.  It's centered around forwarding even though that it only one of the 3 options you're making available.

Possibly consider renaming this to `DivertedTaskHandler`.